pkg/api/http_server.go

```diff
@@ -868,6 +868,20 @@ func handleEncryptedCertificates(cfg *setting.Cfg) (*tls.Certificate, error) {
 		if err != nil {
 			return nil, fmt.Errorf("error marshaling PKCS8 Private key: %w", err)
 		}
+	} else if strings.Contains(keyPemBlock.Type, "RSA PRIVATE KEY") {
+		// Check if the PEM block is encrypted with PKCS#1
+		// This is still requested by some customers (and fairly used)
+		// nolint:staticcheck
+		if x509.IsEncryptedPEMBlock(keyPemBlock) {
```

Comment:
suggest inverting this test, so the happy path continues down and the error path goes to the right

Reply:
fixed in 74d241a9ea120b3acee9f67d621a5d6f350703d5
thanks!